reactive/context_test.ts

```diff
@@ -183,3 +215,486 @@ test("all")("Scope.target works as expected when run within a `with` context", (
   expect(listeners.call).toHaveBeenCalledTimes(1)
   expect(observable.foo).toBe(true)
 })
+
+test("all")("Scope.target works with isolated and shared properties", () => {
+  const { context, listeners } = observe({
+    setOfUrls: new Set<string>(["https://example.com"]),
+    name: "ParentContext",
+  })
+
+  const childContext = context.with({
+    name: "ChildContext",
+    isolatedSetOfUrls: new Set<string>(),
+  })
+
+  // Access the isolated set in the child context
+  childContext.target.isolatedSetOfUrls.add("isolated-url")
+  expect(childContext.target.isolatedSetOfUrls.size).toBe(1)
+  expect(childContext.target.isolatedSetOfUrls.has("isolated-url")).toBe(true)
+
+  // Access the shared set in both contexts
+  childContext.target.setOfUrls.add("https://child.com")
+  expect(context.target.setOfUrls.has("https://child.com")).toBe(true)
+  expect(listeners.set).toHaveBeenCalledTimes(0) // No shared change event is fired
```

Comment:
This comment seems misleading, it claims to look for "no shared `change` event fired" but the code look at `listeners.set`

Also, would it be possible to add some assertions to ensure that `call` events for `Set.add` were called ?

Reply:
Done. Initially I was going to remove the proxy for most native objects including Maps and Sets due to how error prone the previous implementation was, but the need to listen for `add` events is too great, so I've added back support, I also fixed the bugs with the previous implmentation https://github.com/okikio/libs/blob/835b04799b8f5f1b6cd41e66b3fa652460445cfc/reactive/context.ts#L373-L393